quinn/src/tests.rs

```diff
@@ -105,7 +105,9 @@ async fn close_endpoint() {
         .unwrap();
     endpoint.close(0u32.into(), &[]);
     match conn.await {
-        Err(crate::ConnectionError::LocallyClosed) => (),
+        Err(
+            crate::ConnectionError::LocallyClosed | crate::ConnectionError::ApplicationClosed(_),
+        ) => (),
```

Comment:
Do we need to change this? Shouldn't this always return `LocallyClosed`?
What I vaguely understood is that `Connection::closed` and `Connection::close_reason` are supposed to return `ConnectionError`s, while any other functions like the ones on streams, etc. are supposed to return `LocallyClosed` immediately.